File: src/main/java/com/googongill/aditory/external/chatgpt/dto/AutoCategorizeResult.java
package com.googongill.aditory.external.chatgpt.dto;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class AutoCategorizeResult {
    private String title;
    private String summary;
    private String categoryName;

    public static AutoCategorizeResult of(ChatGptResponse chatGptResponse) {
        String jsonString = chatGptResponse.getChoices().get(0).getMessage().getContent();
        ObjectMapper objectMapper = new ObjectMapper();
        JsonNode rootNode = null;
        try {
            rootNode = objectMapper.readTree(jsonString);
        } catch (JsonProcessingException e) {
            throw new RuntimeException(e);
        }
        return AutoCategorizeResult.builder()
                .title(rootNode.get("title").asText())
                .summary(rootNode.get("summary").asText())
                .categoryName(rootNode.get("category").asText())
                .build();
    }
}

File: src/main/java/com/googongill/aditory/external/chatgpt/dto/AutoCategorizeResult.kt
package com.googongill.aditory.external.chatgpt.dto

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.JsonNode
import com.fasterxml.jackson.databind.ObjectMapper

data class AutoCategorizeResult(
    val title: String,
    val summary: String,
    val categoryName: String
) {

    companion object {
        fun of(chatGptResponse: ChatGptResponse): AutoCategorizeResult {
            val jsonString = chatGptResponse.choices[0].message.content
            val objectMapper = ObjectMapper()

            var rootNode: JsonNode = try {
                objectMapper.readTree(jsonString)
            } catch (e: JsonProcessingException) {
                throw RuntimeException(e)
            }
            return AutoCategorizeResult(
                title = rootNode.get("title").asText(),
                summary = rootNode.get("summary").asText(),
                categoryName = rootNode.get("category").asText()
            )
        }
    }

}
